.github/workflows/update-locked-env.yml

```diff
@@ -0,0 +1,100 @@
+    if: ${{ github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch' }}
+    name: Update pinned envs
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup conda
+      uses: conda-incubator/setup-miniconda@v3
+      with:
+        miniforge-version: latest
+        activate-environment: ${{ github.event.repository.name }}
+        channel-priority: strict
+        environment-file: ${{ env.BASE_ENV }}
+
+    - name: Install conda-lock
+      run: |
+        conda install -c conda-forge conda-lock
+
+    - name: Generate lockfiles for all platforms
+      run: |
+        conda-lock -f ${{ env.BASE_ENV }} \
+        -p linux-64 -p osx-64 -p win-64 -p osx-arm64 \
+        -k env --filename-template "envs/{platform}.lock"
+
+        # Rename to .yaml extension
+        for file in envs/*.lock.yml; do
+          mv "$file" "${file%.yml}.yaml"
+        done
+    - name: Add SPDX headers to lock files
+      run: |
+        SPDX_HEADER="# SPDX-FileCopyrightText: PyPSA-Earth and PyPSA-Eur Authors\n#\n# SPDX-License-Identifier: CC0-1.0\n"
+
+        # Add header to all generated lock files
+        for file in envs/*.lock.yaml; do
+          echo "Adding header to $file"
+          echo -e "$SPDX_HEADER" | cat - "$file" > temp && mv temp "$file"
+        done
+
+    - name: Insert environment name in lock files
+      run: |
+        for file in envs/*.lock.yaml; do
+          if [ -f "$file" ]; then
+            echo "Processing $file"
+            if ! grep -q "name: pypsa-earth" "$file"; then
+              # Insert name: pypsa-earth before channels section
+              sed -i '7a name: pypsa-earth' "$file"
+            else
+              echo "name: pypsa-earth already exists in $file"
+            fi
+          fi
+        done
+
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: lockfiles
+        path: envs/*.lock.yaml
+
+  create-pull-request:
+    needs: update-locked-environment
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Download all artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: lockfiles
+        path: envs/
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v7
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        branch: update-locked-environment
+        title: "[github-actions.ci] Update locked envs"
+        body: |
+          Automatically generated PR to update locked environment files for Windows, macOS, and Linux.
+
+          These files were generated using conda-lock for improved dependency resolution and reproducibility.
+
+          **Note: Do not merge without manual test execution. Either update the branch to trigger tests, or use `workflow_dispatch` to run tests manually. Unlike standard PRs, tests will not run automatically.**
```

Comment:
Feels like there is still a room for automation 🙂 

Totally agree that it's not the best idea to merge any PR automatically but probably we can create a temporary branch for running CI tests there. Having a manual steps hampers contributions which change the environment (e.g. our discussion with @danielelerede-oet in #1319). What do you think?

As a quick fix we can probably revise slightly instructions to PR. Happy to do so if you deem it useful.

In any case, that is something which can be addressed in a supplementary PR 🙂 So, no actions needed

Reply:
Many thanks 😄 You mean `Use base env file if it was changed` as a point to be added into the PR template, right?

Reply:
No, in the new CI log there is a section with that name

Reply:
Revised the logging further and also the logic to select the env file; probably it should be fine most of the time

Reply:
Ahh, I see! Very cool automation :D
Also, I confirm that the logging message is clear now :)